common/persistence/elasticsearch/visibility_store.go

```diff
@@ -1020,6 +1028,16 @@ func (s *visibilityStore) parseESDoc(hit *elastic.SearchHit, saTypeMap searchatt
 				logNumberParseError(fieldName, historyLength, err, hit.Id)
 			}
 			record.HistoryLength = historyLengthInt64
+		case searchattribute.ExecutionDuration:
+			var executionDuration json.Number
+			if executionDuration, isValidType = fieldValue.(json.Number); !isValidType {
+				logUnexpectedType(fieldName, fieldValue, hit.Id)
+			}
+			executionDurationInt64, err := executionDuration.Int64()
+			if err != nil {
+				logNumberParseError(fieldName, executionDuration, err, hit.Id)
+			}
+			record.ExecutionDuration = time.Duration(executionDurationInt64)
```

Comment:
do you want to set even if there are errors from above lines? should we break if error happens?

Reply:
It doesn't matter, right? But I removed this code eventually. There is no reason to have `ExecutionDuration` as part of `record` because it always can be easily computed.